operator/pkg/apis/core/v1beta1/configconnector_types.go

```diff
@@ -66,6 +66,34 @@ type ConfigConnectorSpec struct {
 	//+kubebuilder:validation:Enum=Absent;Merge
 	//+kubebuilder:validation:Optional
 	StateIntoSpec *StateIntoSpecValue `json:"stateIntoSpec,omitempty"`
+
+	// ConfigConnector specific experiments
+	Experiments *CCExperiments `json:"experiments,omitempty"`
+}
+
+type CCExperiments struct {
+	// LeaderElection defines the configuration for multi cluster leader election.
+	// +optional
+	LeaderElection *LeaderElectionSpec `json:"leaderElection,omitempty"`
+}
+
+// LeaderElectionSpec defines the configuration for leader election.
+// +optional
+type LeaderElectionSpec struct {
+	// MultiClusterLease defines configuration specific to multi-cluster leader election.
+	// +optional
+	MultiClusterLease *MultiClusterLeaseSpec `json:"multiClusterLease,omitempty"`
+}
+
+// MultiClusterLeaseSpec defines the configuration for a multi-cluster lease.
+type MultiClusterLeaseSpec struct {
+	// The name of the MultiClusterLease object that KCC will create.
+	LeaseName string `json:"leaseName"`
+	// The namespace where the MultiClusterLease object will be created.
+	Namespace string `json:"namespace"`
+	// The unique name of the global lock, which must be shared by all KCC replicas
+	// and workloads across all clusters that are part of the same election.
+	GlobalLockName string `json:"globalLockName"`
```

Comment:
We _could_ allow limited templating in this name, e.g. `gs://my-cluster-locks/kcc/prod/{namespace}`